source/client/options_impl.cc

```diff
@@ -338,9 +367,21 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
   if (uri.isSet()) {
     uri_ = uri.getValue();
   }
+
+  if (h2.isSet() && upstream_protocol.isSet()) {
```

Comment:
Should we log a warning about --h2 being deprecated when set?

Reply:
Done in ce8684efbd63998e1380df0b01bd23876a0f7c9e.